plugins/inputs/mongodb/mongodb_server.go

```diff
@@ -37,6 +37,13 @@ func IsAuthorization(err error) bool {
 	return strings.Contains(err.Error(), "not authorized")
 }
 
+func (s *Server) ping() error {
+	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
```

Comment:
This hard coded 10 seconds might need to be changed. For example, if someone uses this plugin with a 2 second gather and the server is down, ping timeouts will pile up.

Reply:
Do I have a way to get the current configured gather interval ?

Reply:
There's no way for a plugin to get the gather interval.

After seeing this again, I don't think it's a big deal. In my example with 2 second gather interval, there could only be 5 concurrent pings.

I see that you reduced the timeout to 1s which helps too